### ISSUE TITLE
FinalLocalVariable doesn't report variable when first assignment is separated

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#1551]
Example:
````
public class TestClass3 {
	public static void test() {
		String test;
		test = "123";
		if (test != null) {}
	}
}
````

Config:
````
<module name="Checker">
	<property name="charset" value="ASCII" />
	<property name="severity" value="error" />
	<property name="fileExtensions" value="java" />

	<module name="TreeWalker">
		<module name="FinalLocalVariable" />
	</module>
</module>
````

Checkstyle reports no errors even though the variable is only assigned once and can be given the "final" modifier. It should report the error on the line declaring the "test" variable.
This fails because Checkstyle doesn't count the number of assignments. It removes the variable on the first assignment it finds, and doesn't realize that the variable declaration gave no initial assignment.
Source: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java#L201

**Comment by romani** [comment]
PR #1722 is non completed

**Comment by romani** [comment]
merged.
